### DISCUSSION TITLE
"dbt docs generate" throws error on macro definition.

### DISCUSSION THREAD

**Discussion by ghkgupta** [ScalefreeCOM/datavault4dbt#264] [Q&A] [answered]
I have nearly 2000 dbt models (incremental, non-incremental & views). Upon giving "dbt docs generate" command to view the model graph, I get the following error.

`
Runtime Error
  Compilation Error in model stg__030__af1 (models/20_stage/030/orders/stg__030__af1 .sql)
    Could not find the derived_column input column PROPERTY inside the source relation dev.DBT.raw__030__af1

> in macro default__derived_columns_datatypes (macros/staging/derived_column_datatypes.sql)
    > called by macro derived_columns_datatypes (macros/staging/derived_column_datatypes.sql)
    > called by macro snowflake__stage (macros/staging/snowflake/stage.sql)
    > called by macro stage (macros/staging/stage.sql)
    > called by model stg__030__af1 (models/20_stage/030/orders/stg__030__af1.sql)
Exception ignored in: <function Pool.__del__ at 0x7f95f4ebc160>
`
`raw__030__af1`  --> `stg__030__af1`

However, when I build and run the raw model `raw__030__af1`, it completes successfully without requiring any changes to the **raw** and **stage** dbt models. The issue arises when I run **dbt docs generate** as a similar error occurs with a different model each time. This continues until all models are built, which includes around 2000 models, whereas the models I'm currently working on constitute only a small subset. Consequently, I have to execute those models on large datasets, which significantly increases the runtime.

Here is the second error after building `raw__030__af1`  --> `stg__030__af1`
`
Runtime Error
  Compilation Error in model stg__030__kf1 (models/20_stage/030/orders/stg__030__kf1 .sql)
    Could not find the derived_column input column CUSTOMER inside the source relation dev.DBT.raw__030__kf1

> in macro default__derived_columns_datatypes (macros/staging/derived_column_datatypes.sql)
    > called by macro derived_columns_datatypes (macros/staging/derived_column_datatypes.sql)
    > called by macro snowflake__stage (macros/staging/snowflake/stage.sql)
    > called by macro stage (macros/staging/stage.sql)
    > called by model stg__030__kf1 (models/20_stage/030/orders/stg__030__kf1.sql)
Exception ignored in: <function Pool.__del__ at 0x7f95f4ebc160>
`
`raw__030__kf1`  --> `stg__030__kf1`

**Is there any way to solve this problem?**

**Comment by tkiehn** [comment, marked as answer]
Hi @ghkgupta,

this seems to be a similar issue like in [this discussion](https://github.com/ScalefreeCOM/datavault4dbt/discussions/225).

The proposed solution there is to first do a compile against a target where all objects are available, then do a
dbt docs generate --no-compile.

Let me know if this helps!

Kind regards
Theo

**Reply by ghkgupta** [reply]
Hi @tkiehn,

It helps a lot. Thank you very much.

Br,
Hari Gudla
